### ISSUE TITLE
Typescript config: allow disabling darkMode

### ISSUE THREAD

**Issue by skyqrose** [tailwindlabs/tailwindcss#10914]
TL;DR: In the config types, allow `darkMode: false`.

**What version of Tailwind CSS are you using?**

For example: v3.3.0

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss 8.4.21

**What version of Node.js are you using?**

node 18.15.0, typescript 5.0.2

**What browser are you using?**

N/A

**What operating system are you using?**

macOS

**Reproduction URL**

described below

**Describe your issue**

We disable tailwind's dark mode and have our own plugin to control the `dark:` modifier, so we can specify `dark:` and `light:` modifiers symmetrically as `class="light:bg-white dark:bg-black"` instead of `class="bg-white dark:bg-black"`.

```ts
// myapp/tailwind.config.ts
import { Config } from "tailwindcss";
import plugin from "tailwindcss/plugin";

const lightModePlugin = plugin(({ addVariant }) => {
  addVariant("light", ":where(html.light) &");
  addVariant("dark", [
    ":where(html.dark) &",
    ":where(html:not(.light):not(.dark)) &",
  ]);
});

module.exports = {
  // ...
  // setting to neither valid value disables built-in dark mode entirely
  darkMode: null as unknown as "media" | "class",
  plugins: [lightModePlugin],
} satisfies Config;
```

According to the [getting started docs](https://tailwindcss.com/docs/upgrade-guide#remove-dark-mode-configuration), you can set `darkMode: false` to disable it. However, the published types for the `darkMode` option don't allow that. (See the excerpt from the `.d.ts` file below.)

So we had to do the typescript `as unknown as DarkModeConfig` in order to work around the omission.

Relatedly, the main [dark mode docs](https://tailwindcss.com/docs/dark-mode) don't mention this `darkMode: false` feature, and they probably should.

```ts
// tailwindcss/types/config.d.ts
// DarkMode related config
type DarkModeConfig =
  // Use the `media` query strategy.
  | 'media'
  // Use the `class` strategy, which requires a `.dark` class on the `html`.
  | 'class'
  // Use the `class` strategy with a custom class instead of `.dark`.
  | ['class', string]

interface RequiredConfig {
  // ...
}

interface OptionalConfig {
  // ...
  darkMode: Partial<DarkModeConfig>
  // ...
  plugins: Partial<PluginsConfig>
  // Custom
  [key: string]: any
}

export type Config = RequiredConfig & Partial<OptionalConfig>
```

**Comment by adamwathan** [comment]
Hey! I think you’re misreading those docs — as of v3 there is no `false` option, that’s why the docs say to remove that config if it’s currently set to false.

We originally supported `false` because Tailwind didn’t generate styles on demand before just-in-time mode (which is now the only mode) and we wanted to give people a way to avoid generating CSS they didn’t need, but that’s not needed now. By overriding the `dark` variant with a custom one you are disabling the built-in one anyways, so you just don’t need any dark mode config at all, just remove it 👍🏻

**Comment by thomasgauvin** [comment]
+1, it would be important for people that want to use tailwind but want to opt out of having a dark mode to be able to set this in the config

**Comment by yp717** [comment]
+1 I agree this would also be useful for our team

**Comment by yp717** [comment]
After several hours of investigation we have figured out how to do this 🎉  It seems that unless you provide an alternative string for the value of "darkMode" in the tailwind config, it falls back to Media. Because Tailwind does not let you operate on the "dark:" prefixes, the easiest way to accomplish this is to set the "darkMode" value to another string - don't leave it blank, don't set it to class, but set it to something else entirely and then you can hijack the behaviour of the dark variant 😄 

If you want to see an example of what this looks like I've used this in the context of this plugin that pulls together radix/tailwind - the plugin may not be what you want but it shows you to "disable" dark mode: https://github.com/yp717/tailwindcss-radix-colors-plugin @skyqrose @thomasgauvin
